src/renderers/controls/date.control.tsx

```diff
@@ -21,14 +21,14 @@ export const dateControlTester: RankedTester = rankWith(2, and(
 export class DateControl extends Control<ControlProps, ControlState> {
 
   render() {
-    const { classNames, id, visible, enabled, errors, label, uischema } = this.props;
+    const { classNames, id, visible, enabled, errors, label, uischema, required } = this.props;
     const isValid = errors.length === 0;
     const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
 
     return (
       <div className={classNames.wrapper}>
         <label htmlFor={id} className={classNames.label} data-error={errors}>
-          {label}
+          {required ? label + '*' : label}
```

Comment:
please extract this to a util to remove code duplication

Reply:
a good place would be the renderer.util

Reply:
should I remove `required` field from `props`

Reply:
So you want me to create a function in renderer.util, so controllers will call this function with the `required` parameter. Am I correct?

Reply:
@AhmetTanakol Yes, e.g. think about what would be necessary if you change the asterisk to a different character: You'd need to change the line in each renderer, which is cumbersome.